bin/oref0-ns-autotune.sh

```diff
@@ -0,0 +1,173 @@
+    ;;
+    -e=*|--end-date=*)
+    END_DATE="${i#*=}"
+    END_DATE=`date --date="$END_DATE" +%Y-%m-%d`
+    shift # past argument=value
+    ;;
+    -r=*|--runs=*)
+    NUMBER_OF_RUNS="${i#*=}"
+    shift # past argument=value
+    ;;
+    -x=*|--xlsx=*)
+    EXPORT_EXCEL="${i#*=}"
+    shift # past argument=value
+    ;;
+    *)
+    # unknown option
+    echo "Option ${i#*=} unknown"
+    UNKNOWN_OPTION="yes"
+    ;;
+esac
+done
+
+if [[ -z "$DIR" || -z "$NIGHTSCOUT_HOST" ]]; then
+    echo "Usage: oref0-ns-autotune <--dir=openaps_directory> <--ns-host=https://mynightscout.azurewebsites.net> [--start-date=YYYY-MM-DD] [--end-date=YYYY-MM-DD] [--runs=number_of_runs] [--xlsx=autotune.xlsx]"
+exit 1
+fi
+if [[ -z "$START_DATE" ]]; then
+    # Default start date of yesterday
+    START_DATE=`date --date="1 day ago" +%Y-%m-%d`
+fi
+if [[ -z "$END_DATE" ]]; then
+    # Default end-date as this morning at midnight in order to not get partial day samples for now
+    # (ISF/CSF adjustments are still single values across each day)
+    END_DATE=`date --date="1 day ago" +%Y-%m-%d`
+fi
+
+if [[ -z "$UNKNOWN_OPTION" ]] ; then # everything is ok
+  echo "Running oref0-autotune-test.sh --dir=$DIR --ns-host=$NIGHTSCOUT_HOST --start-date=$START_DATE --runs=$NUMBER_OF_RUNS --end-date=$END_DATE"
+else
+  echo "Unknown options. Exiting"
+  exit 1
+fi
+
+
+# Get profile for testing copied to home directory. "openaps" is my loop directory name.
+cd $directory && mkdir -p autotune
+cp settings/pumpprofile.json autotune/profile.pump.json
+# If a previous valid settings/autotune.json exists, use that; otherwise start from settings/profile.json
+cp settings/autotune.json autotune/profile.json && cat autotune/profile.json | json | grep -q start || cp autotune/profile.pump.json autotune/profile.json
+cd autotune
+# TODO: Need to think through what to remove in the autotune folder...
+
+# Pull Nightscout Data
+echo "Grabbing NIGHTSCOUT treatments.json for date range..."
+
+# Get Nightscout carb and insulin Treatments
+curl "$NIGHTSCOUT_HOST/api/v1/treatments.json?find\[created_at\]\[\$gte\]=`date -d $START_DATE -Iminutes`&\[\$lte\]=`date -d $END_DATE -Iminutes`" > ns-treatments.json
+
+# Build date list for autotune iteration
+date_list=()
+date=$START_DATE; 
+while :
+do 
+  date_list+=( "$date" )
+  if [ $date != "$END_DATE" ]; then 
+    date="$(date --date="$date + 1 days" +%Y-%m-%d)"; 
+  else 
+    break
+  fi
+done
+
+echo "Grabbing NIGHTSCOUT entries/sgv.json for date range..."
+
+# Get Nightscout BG (sgv.json) Entries
+for i in "${date_list[@]}"
+do 
+  curl "$NIGHTSCOUT_HOST/api/v1/entries/sgv.json?find\[date\]\[\$gte\]=`(date -d $i +%s | tr -d '\n'; echo 000)`&find\[date\]\[\$lte\]=`(date --date="$i +1 days" +%s | tr -d '\n'; echo 000)`&count=1000" > ns-entries.$i.json
+done
+
+echo "Running $NUMBER_OF_RUNS runs from $START_DATE to $END_DATE"
+sleep 2
+
+# Do iterative runs over date range, save autotune.json (prepped data) and input/output 
+# profile.json
```

Comment:
Shouldn't we use a different file name as `profile.json`, e.g. `profile.autotune.js` and 
`cp profile.json profile.autotune.js` at the start of autotune and `cp profile.autotune.js profile.json` at then end of autotune, if the users want's to automatically tune profiles. This will be a simple way of implementing https://github.com/openaps/oref0/issues/99 

I would advise the default is only to generate  autotune files, and not to use the `profile.autotune.json` by default